### PR TITLE
Switch cloud-on-k8s docs to asciidoctor

### PR DESCRIPTION
**Pull request by nik9000**
Switches the cloud-on-k8s docs from the unmaintained `AsciiDoc` tool to
the actively maintained `Asciidoctor` tool.

### CODE REVIEW

File: doc_build_aliases.sh [hunk 1]
```diff
@@ -75,7 +75,7 @@ alias docbldece='$GIT_HOME/docs/build_docs --doc $GIT_HOME/cloud/docs/cloud-ente
 # Cloud
 alias docbldech='$GIT_HOME/docs/build_docs --doc $GIT_HOME/cloud/docs/heroku/index.asciidoc --resource=$GIT_HOME/cloud/docs/shared --resource=$GIT_HOME/cloud/docs/saas --chunk 1'
 
-alias docbldk8s='$GIT_HOME/docs/build_docs --doc $GIT_HOME/cloud-on-k8s/docs/index.asciidoc --resource $GIT_HOME/cloud/docs/shared --chunk 1'
+alias docbldk8s='$GIT_HOME/docs/build_docs --asciidoctor --doc $GIT_HOME/cloud-on-k8s/docs/index.asciidoc --resource $GIT_HOME/cloud/docs/shared --chunk 1'
```
Comment (nik9000) [inline doc_build_aliases.sh]: It looks like you weren't using the `cloud/docs/shared` resource.

**Comment by nik9000** [comment]
This changes a few ids:
* k8s-monitor_cluster_health_and_creation_progress -> k8s_monitor_cluster_health_and_creation_progress
* k8s-request_elasticsearch_access -> k8s_request_elasticsearch_access
* k8s-a_single_elasticsearch_node_per_kubernetes_host_default -> k8s_a_single_elasticsearch_node_per_kubernetes_host_default
* k8s-local_persistent_volume_constraints -> k8s_local_persistent_volume_constraints
* k8s-node_affinity -> k8s_node_affinity
* k8s-kubernetes_distribution_support -> k8s_kubernetes_distribution_support
* k8s-apm_server -> k8s_apm_server
* k8s-user_provided_http_certificates -> k8s_user_provided_http_certificates
* k8s-user_provided_pod_templates -> k8s_user_provided_pod_templates
* k8s-breaking_changes -> k8s_breaking_changes

It looks like Asciidoctor is automatically switching the `-` in the `idprefix` to a `_` which is weird. I *think* it is ok, but I'll defer anyone with a strong opinion.

**Comment by nik9000** [comment]
[Here](http://docs_1030.docs-preview.app.elstc.co/guide/en/cloud-on-k8s/index.html) is a preview, just in case anyone wants one.

**Comment by alaudazzi** [comment]
FYI @artemnikitin and related to https://github.com/elastic/k8s-dev/issues/27.

File: doc_build_aliases.sh (hunk 1 above)
Comment (alaudazzi) [reply doc_build_aliases.sh]: Not yet, but we might need in the future. Actually, this is something I'd like to discuss with one of you two (@nik9000 or @nrichers)

**Pull request merged**